lib/matplotlib/ticker.py

```diff
@@ -2256,7 +2254,11 @@ def tick_values(self, vmin, vmax):
             # If we're a minor locator *that expects at least two ticks per
             # decade* and the major locator stride is 1 and there's no more
             # than one minor tick, switch to AutoLocator.
-            return AutoLocator().tick_values(vmin, vmax)
+            ticklocs = AutoLocator().tick_values(vmin, vmax)
+            # Don't overstrike the major labels.
+            ticklocs = ticklocs[
+                ~is_close_to_int(np.log(ticklocs) / np.log(b))]
```

Comment:
This looks like a hard-coded assumption on the position of the major ticks. Will this fail if I’ve set non-default values no the major locator?

Reply:
Absolutely.  See discussion above for fixing that.  But that assumption is already in the code, this just makes that assumption work properly under all conditions.  The next step is to get rid of the assumption, but thats a bigger project.

Reply:
Ok, that's reasonable for now. However, it would be good to explicitly document this in the code with a comment (unless you're immediately following up with the full fix).

Reply:
Sure thing